pkg/server/test/list_objects.go

```diff
@@ -297,7 +321,9 @@ func TestListObjectsRespectsMaxResults(t *testing.T, ds storage.OpenFGADatastore
 
 				require.NotNil(t, res)
 				require.NoError(t, err)
-				require.LessOrEqual(t, len(res.Objects), int(test.maxResults))
+				if test.maxResults != 0 { // don't get all results
```

Comment:
If `text.maxResults == 0` then we should just assert they are exactly equal (ignoring order). Otherwise, then we can assert that the length of the results is less than or equal to max results AND greather than or equal to minimum results. That is,

```go
if test.maxresults == 0 {
    require.ElementsMatch(...)
} else {
    require.LessOrEqual(t, len(res.Objects), int(test.maxResults))
    require.GreaterOrEqual(t, len(res.Objects), int(test.minimumResultsExpected))
    require.Subset(t, test.allResults, res.Objects)
}
```